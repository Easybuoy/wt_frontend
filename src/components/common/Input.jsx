import React from "react";
import { Input as StyledInput } from "@chakra-ui/core";
import styled from "styled-components";

const CustomStyledInput = styled.div`
  margin: 1rem 0;

  span {
    color: red;
  }
`;

function Input({
  id,
  name,
  placeholder,
  variant,
  type,
  onChange,
  value,
  error
}) {
  return (
    <CustomStyledInput>
      <StyledInput
        id={id}
        name={name}
        placeholder={placeholder}
        variant={variant}
        type={type}
        onChange={onChange}
        value={value}
        bg="#FFFCF2"
        _hover="black"
        focusBorderColor="#FF8744"
        errorBorderColor="crimson"
      />
      <span>{error}</span>
    </CustomStyledInput>
  );
}

export default Input;
