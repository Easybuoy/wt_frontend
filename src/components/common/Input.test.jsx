import React from "react";
import { shallow } from "enzyme";

import Input from "./Input";

describe("<Input />", () => {
  let wrapper;

  const props = {
    location: { search: jest.mock() },
    client: { mutate: jest.fn(), query: jest.fn() },
    history: { push: jest.fn() },
    id: "search",
    name: "search",
    placeholder: "Search",
    variant: "jest.mock()",
    type: "text",
    onChange: jest.fn(),
    value: "",
    error: ""
  };

  beforeEach(() => {
    wrapper = shallow(<Input {...props} />);
  });

  it("renders the Input component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
