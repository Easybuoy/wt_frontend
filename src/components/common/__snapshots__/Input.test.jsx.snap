// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders the Input component correctly 1`] = `
<styled.div>
  <Input
    _hover="black"
    as="input"
    bg="#FFFCF2"
    errorBorderColor="crimson"
    focusBorderColor="#FF8744"
    id="search"
    isFullWidth={true}
    name="search"
    onChange={[MockFunction]}
    placeholder="Search"
    size="md"
    touchedName={true}
    type="text"
    value=""
    variant="jest.mock()"
  />
</styled.div>
`;
